crates/core/src/sql/compiler.rs

```diff
@@ -248,7 +248,7 @@ fn compile_drop(name: String, kind: DbType, table_access: StAccess) -> Result<Cr
 }
 
 /// Compiles a `SQL` clause
-fn compile_statement(statement: SqlAst) -> Result<CrudExpr, PlanError> {
+fn compile_statement(db: &RelationalDB, statement: SqlAst) -> Result<CrudExpr, PlanError> {
```

Comment:
Are we sure we'd like to pass the entire DB here if we're just using the address? Is the idea that it will eventually be needed when the metrics are in the DB instead of prometheus?

Reply:
We can definitely get away with just the address for now, but yes we will need the db once we start storing a full set of statistics for query optimization.